apps/sel4test-tests/src/tests/scheduler.c

```diff
@@ -1695,3 +1695,72 @@ static int test_changing_affinity(struct env *env)
 }
 DEFINE_TEST(SCHED0022, "test changing a helper threads core", test_changing_affinity,
             (config_set(CONFIG_KERNEL_MCS) &&(CONFIG_MAX_NUM_NODES > 1)));
+
+
+/* busy wait for a number of iterations */
+static void delay_loop(unsigned iterations) {
+    for (int i = 0; i < iterations; i++) {
+        __asm__ __volatile__ ("");
+    }
+}
+
+#define SCHED0023_NUM_ROUNDS 5
+
+/* used in test_round_robin below */
+static void round_robin_thread(int id, volatile unsigned long *counters)
+{
+    delay_loop(id * 100);
```

Comment:
Why do we need this delay actually, is it to ensure `counters` is initialized and visible?

Reply:
It’s to make sure that the writes to the counters are in order. When the threads are first started, this is not necessary, but when each thread gets scheduled the second time, we’d be relying on that each of them got the exact same number of cycles to make progress. If thread 0 got just 10 cycles less, the second counter increment in thread 0 might happen after the second counter increment in thread 1. I don’t think the timer interrupts are that accurate for all platforms, so I’m adding some safety.

Reply:
Good point. Please add this in a comment in the code.